packages/grid/x-data-grid/src/joy/joySlots.tsx

```diff
@@ -45,14 +97,103 @@ const TextField = React.forwardRef<
   );
 });
 
+const Button = React.forwardRef<
+  HTMLButtonElement,
+  NonNullable<GridSlotsComponentsProps['baseButton']>
+>(function Button({ startIcon, color, endIcon, size, sx, variant, ...props }, ref) {
+  return (
+    <JoyButton
+      {...props}
+      size={convertSize(size)}
+      color={convertColor(color)}
+      variant={convertVariant(variant) ?? 'plain'}
+      ref={ref}
+      startDecorator={startIcon}
+      endDecorator={endIcon}
+      sx={sx as SxProps<Theme>}
+    />
+  );
+});
+
+const IconButton = React.forwardRef<
```

Comment:
Maybe make icon buttons smaller?
<img width="352" alt="Screenshot 2023-04-24 at 18 04 09" src="https://user-images.githubusercontent.com/13808724/234053813-45567e1c-b77d-4588-a3fd-6775604d857b.png">

Reply:
Sure! Material UI uses 30px (Joy UI is 32px).